File: beginner/c/1186.c
#include <stdio.h>

int main (){

  double m[12][12], soma = 0.0;
  int i, j, cont = 0;
  char operacao;

  scanf(" %c", &operacao);

  for(i = 0; i < 12; i++){
    for(j = 0; j < 12; j++){
      scanf("%lf", &m[i][j]);
    }
  }

  int coluna = 11;
  for(i = 0; i < 12; i++){
    for(j = coluna; j > coluna - i; j--){
        soma += m[i][j];
        cont++;
    }
  }

  if(operacao == 'S'){
    printf("%.1lf\n", soma);
  } else{
    printf("%.1lf\n", soma/cont);
  }


}

File: beginner/python/1186.py
operacao = input()

m = [[0 for j in range(12)] for i in range(12)]

for i in range(12):
    for j in range(12):
        m[i][j] = float(input())

def soma_or_media(mat, op):
    coluna = 11
    cont = 0
    soma = 0.0

    for i in range(12):
        for j in range(coluna, coluna - i, -1):
            soma += mat[i][j]
            cont += 1

    if op == 'S':
        return soma
    else:
        return soma/cont

print("%.1f" % (soma_or_media(m, operacao)))